NetworkPkg/Ip6Dxe/Ip6Nd.c

```diff
@@ -1169,7 +1176,14 @@ Ip6SendRouterSolicit (
                                                  sizeof (IP6_ETHER_ADDR_OPTION),
                                                  FALSE
                                                  );
-    ASSERT (LinkLayerOption != NULL);
+    // MU_CHANGE [BEGIN] - CodeQL change
+    if (LinkLayerOption == NULL) {

```

Comment:
Looks like `NetBufFree (IcmpHead);` should be in this block.

Reply:
These `NetbufAllocSpace()` calls appear to be carving out space from an already allocated buffer so I'll drop my comment on them.